pkg/controller/agent/linkmonitor/controller.go

```diff
@@ -185,35 +175,34 @@ func (h Handler) AddPattern(lm *networkv1.LinkMonitor) error {
 	return nil
 }
 
-func (h Handler) DeletePattern(lm *networkv1.LinkMonitor) {
-	h.linkMonitor.DeletePattern(lm.Name)
-}
+func compareLinkStatusList(m, n []networkv1.LinkStatus) bool {
+	if len(m) != len(n) {
+		return false
+	}
 
-func (h Handler) isRuleChange(lm *networkv1.LinkMonitor) bool {
-	pattern, ok := h.linkMonitor.ListPatterns()[lm.Name]
-	if ok {
-		if pattern.Type == lm.Spec.TargetLinkRule.TypeRule && pattern.Name == lm.Spec.TargetLinkRule.NameRule {
+	if (m == nil) != (n == nil) {
```

Comment:
Is this case also belonging to `if len(m) != len(n)` ?

Reply:
Yes, good catch.